From ce75c59f23289b8765673233bac7052ccf850752 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Wed, 1 Jan 2020 10:17:49 -0800
Subject: [PATCH] CMakeLists.txt: Drop 2nd call to find_package Boost

* fixes build with Boost 1.72.0 where it was failing, because this call has
  Boost_USE_MULTITHREADED OFF
  and the later call has
  Boost_USE_MULTITHREADED ON

-- Adding -DWEBOS_TARGET_MACHINE_QEMUX86
CMake Error at /jenkins/mjansa/build-ros/webos-dashing-master/tmp-glibc/work/qemux86-webos-linux/audiod/1.0.0-12-r25/recipe-sysroot/usr/lib/cmake/Boost-1.72.0/BoostConfig.cmake:120 (find_package):
  Found package configuration file:

    /jenkins/mjansa/build-ros/webos-dashing-master/tmp-glibc/work/qemux86-webos-linux/audiod/1.0.0-12-r25/recipe-sysroot/usr/lib/cmake/boost_regex-1.72.0/boost_regex-config.cmake

  but it set boost_regex_FOUND to FALSE so package "boost_regex" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_regex.so.1.72.0 (multithreaded, Boost_USE_MULTITHREADED=OFF)

  * libboost_regex.a (static, Boost_USE_STATIC_LIBS=OFF)

Call Stack (most recent call first):
  /jenkins/mjansa/build-ros/webos-dashing-master/tmp-glibc/work/qemux86-webos-linux/audiod/1.0.0-12-r25/recipe-sysroot/usr/lib/cmake/Boost-1.72.0/BoostConfig.cmake:185 (boost_find_component)
  /jenkins/mjansa/build-ros/webos-dashing-master/tmp-glibc/work/qemux86-webos-linux/audiod/1.0.0-12-r25/recipe-sysroot-native/usr/share/cmake-3.16/Modules/FindBoost.cmake:443 (find_package)
  CMakeLists.txt:35 (find_package)

-- Configuring incomplete, errors occurred!
See also "/jenkins/mjansa/build-ros/webos-dashing-master/tmp-glibc/work/qemux86-webos-linux/audiod/1.0.0-12-r25/build/CMakeFiles/CMakeOutput.log".

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c800a07..e983a26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,13 +28,6 @@ webos_add_compiler_flags(ALL "-Wall")
 
 webos_machine_dep()
 
-# These variables can change the default behaviour and need to be set before calling find_package
-set(Boost_USE_STATIC_LIBS OFF)
-set(Boost_USE_MULTITHREADED OFF)
-set(Boost_USE_STATIC_RUNTIME OFF)
-find_package(Boost REQUIRED COMPONENTS regex system filesystem)
-include_directories(Boost_INCLUDE_DIRS)
-
 include(FindPkgConfig)
 
 pkg_check_modules(GLIB2 REQUIRED glib-2.0)
-- 
2.17.1

